pkg/protocol/handler.go

```diff
@@ -128,6 +128,11 @@ func (h *Handler) Update(msg *Message) {
 		return
 	}
 
+	h.store(msg)
```

Comment:
putting the `store` here solves an issue where the handler would store the P2P message in the queue, then advance when the round changed. The `verifyMessage` method would exit early since it assumed the broadcast message was not available, but `finalize` would still proceed.